packages/react/src/components/Table/TableToolbar/TableToolbar.jsx

```diff
@@ -284,17 +310,49 @@ const TableToolbar = ({
         totalSelected={totalSelected}
         translateWithId={(...args) => tableTranslateWithId(i18n, ...args)}
       >
-        {batchActions.map(({ id, labelText, ...others }) => (
+        {visibleBatchActions.map(({ id, labelText, disabled, ...others }) => (
           <TableBatchAction
             key={id}
             onClick={() => onApplyBatchAction(id)}
             tabIndex={shouldShowBatchActions ? 0 : -1}
-            disabled={!shouldShowBatchActions}
+            disabled={!shouldShowBatchActions || disabled}
             {...others}
           >
             {labelText}
           </TableBatchAction>
         ))}
+        {hasVisibleOverflowBatchActions ? (
+          <OverflowMenu
+            data-testid={`${testID || testId}-batch-actions-overflow-menu`}
+            className={`${iotPrefix}--table-overflow-batch-actions`}
+            flipped={langDir === 'ltr'}
+            direction="bottom"
+            onClick={(e) => e.stopPropagation()}
+            renderIcon={OverflowMenuVertical20}
+            tabIndex={shouldShowBatchActions ? 0 : -1}
+            size="lg"
+            menuOptionsClass={`${iotPrefix}--table-overflow-batch-actions__menu`}
+          >
+            {visibleOverflowBatchActions.map(
+              ({ id, labelText, disabled, hasDivider, isDelete, renderIcon }) => (
+                <OverflowMenuItem
+                  data-testid={`${testID || testId}-batch-actions-overflow-menu-item-${id}`}
+                  itemText={renderTableOverflowItemText({
+                    action: { renderIcon, labelText },
+                    className: `${iotPrefix}--table-toolbar-aggregations__overflow-menu-content`,
+                  })}
+                  disabled={!shouldShowBatchActions || disabled}
+                  onClick={() => onApplyBatchAction(id)}
+                  key={`table-batch-actions-overflow-menu-${id}`}
+                  requireTitle={!renderIcon}
```

Comment:
What does this actually do? The items with `renderIcon` still gets a hover title...

Reply:
It adds the `title` to items that _don't_ have an icon. It's for blocks of text that might be long and get truncated.

Reply:
Ok, so we have to "require" the title unless there is a renderIcon in which case there will always be a title added?

Reply:
Yes, the renderIcon will always add a title. The underlying OverflowMenuItem will not unless `requireTitle` is true. Then, it'll use either `itemText || title` as the title. Instead of measuring the text for truncation we just set `requireTitle={!renderIcon}` to always force carbon to add the title for items that are text only.